### ISSUE TITLE
How to change disabled input text value based on radio buttons click?

### ISSUE THREAD

**Issue by jinxac** [erikras/redux-form#1974]
I have 3 radio buttons in my form and an input text field. 

On clicking the first radio button option the value in text field should be set  as 25000.
On clicking the second radio button option the value in text field should be set  as 50000.
On clicking the third radio button option the value in text field should be set  as 75000.

My Input text field is a disabled text field so I cant edit them actually.

I tried using [normalizing](http://redux-form.com/6.1.1/examples/normalizing/) but this works only when a field is edited. Please correct me if I am wrong. Is there any way I can change the value of input when I call normalize for radio button?

How do I achieve this?

**Comment by JakeLin1024** [comment]
Is using Selector or formValueSelector a workaround?

**Comment by bmv437** [comment]
you could set the value of the radio buttons to be equal to 25000, 50000, and 75000 respectively. Then just put a text input to that value of the form. (disclaimer, I just whipped this up, it's not guaranteed to work)

```javascript
const RadioGroup = ({input, options}) => (
  <div>
    {options.map(({label, value}, index) => (
      <div>
        <input type='radio' {...input} value={value} id={`${input.name}-${value}`} checked={value === input.value} />
        <label htmlFor={`${input.name}-${value}`}>{label}</label>
      </div>
    )}
  </div>
)

. . .

<Field name="money" component={RadioGroup} options={[
  { label: "small", value: 25000 },
  { label: "medium", value: 50000 },
  { label: "large", value: 75000 },
]} />

<Field name="money" component="text" disabled />
```

**Comment by jinxac** [comment]
@bmv437 : this wont work i guess because i already have a value assigned to radio button which i need to pass in my API call.

**Comment by jinxac** [comment]
@erikras any help on this please?

**Comment by erikras** [comment]
So you have a field, we'll call it `salary`, that you want to update based on some other radio field, we'll call it `experience`, but the `experience` field has its own value (e.g. `0 years`, `3 years`, `6 years`)  that you want to save, and the `salary` field is never editable. Is this correct?

If that's correct, then `salary` is not really a field. It's a _calculated property_ based on the other state of the form. You can render it like a disabled input if you'd like, by injecting a [`formValueSelector`](http://redux-form.com/6.2.0/docs/api/FormValueSelector.md/) value of your `experience` field:

```jsx
const toSalary = experience => {
  switch(experience) {
    case '0 years':
      return 25000
    case '3 years':
      return 50000
    case '6 years':
      return 75000
  }
}
...
<input type="text" disabled value={toSalary(this.props.experience)}/>
```
Then your onSubmit could use the same `toSalary()` function to calculate the other value.
```jsx
onSubmit = values =>
  ajax.post('/saveWidget', {
    ...values,
    salary: toSalary(values.experience)
  })
```
